tests/fsharp/Libraries/ControlTests.fs

```diff
@@ -0,0 +1,176 @@
+﻿namespace FSharp.Libraries.UnitTests
+
+open System
+open NUnit.Framework
+open FSharp.Compiler.UnitTests
+
+[<TestFixture>]
+module ControlTests =
+    // Regression for FSHARP1.0:5969
+    // Async.StartChild: error when wait async is executed more than once
+    [<Test>]
+    let ExecuteAsyncMultipleTimes() =
+        CompilerAssert.CompileExeAndRun
+            """
+module M
+
+let a = async {
+                let! a = Async.StartChild(
+                            async {
+                                do! Async.Sleep(500)
+                                return 27
+                            })
+                let! result  = Async.Parallel [ a; a; a; a ]
+                return result
+            } |> Async.RunSynchronously
+
+exit 0
+            """
+
+
+    // Regression for FSHARP1.0:5970
+    // Async.StartChild: race in implementation of ResultCell in FSharp.Core
+    [<Test>]
+    let JoiningStartChild() =
+        CompilerAssert.CompileExeAndRun
+            """
+module M
+
+let Join (a1: Async<'a>) (a2: Async<'b>) = async {
+      let! task1 = a1 |> Async.StartChild
+      let! task2 = a2 |> Async.StartChild
+      
+      let! res1 = task1
+      let! res2 = task2 
+      return (res1,res2) }
+
+let r =
+    try 
+                     Async.RunSynchronously (Join (async { do! Async.Sleep(30) 
+                                                           failwith "fail"
+                                                           return 3+3 }) 
+                                                  (async { do! Async.Sleep(30) 
+                                                           return 2 + 2 } ))
+    with _ -> 
+                     (0,0)     
+
+exit 0
+
+            """
+    // Regression test for FSHARP1.0:6086
+    // This is a bit of duplication because the same/similar test
+    // can also be found under the FSHARP suite. Yet, I like to have
+    // it here...
+    
+    // The interesting thing about this test is that is used to throw
```

Comment:
This comment can probably also be removed; running this as a console app on x64 .NET Core does not throw